src/main/java/com/arpnetworking/tsdcore/sinks/HttpSinkActor.java

```diff
@@ -257,12 +274,22 @@ private void dispatchPending() {
     }
 
     private void fireNextRequest() {
-        final Request request = _pendingRequests.poll();
+        final Entry<Request, Long> requestEntry = _pendingRequests.poll().entrySet().asList().get(0);
+        final Metrics metrics = _metricsFactory.create();
+        metrics.setTimer(_inQueueLatencyName, requestEntry.getValue() - System.currentTimeMillis(), TimeUnit.MILLISECONDS);
+
+        final Request request = requestEntry.getKey();
         _inflightRequestsCount++;
 
         final CompletableFuture<Response> promise = new CompletableFuture<>();
+        metrics.startTimer(_requestLatencyName);
         _client.executeRequest(request, new ResponseAsyncCompletionHandler(promise));
         final CompletionStage<Object> responsePromise = promise
+                .whenComplete((result, err) -> {
+                        metrics.stopTimer(_requestLatencyName);
+                        metrics.incrementCounter(_requestSuccessName, err == null ? 1 : 0);
```

Comment:
This only treats exceptions as failures. What about non-success http status codes in `processCompletedRequest` which is only fired after this method? I think you need to do some refactoring here.